src/rpc/blockchain.cpp

```diff
@@ -1757,6 +1775,7 @@ static UniValue getblockstats(const UniValue &params, bool fHelp)
     }
 
     const CBlock block = GetBlockChecked(pindex);
+    const CBlockUndo blockUndo = GetUndoChecked(pindex);
 
```

Comment:
let's check that blockundo.vtxundo.size() >= block.vtx.size()-1, and throw a JSONRPCError "Corrupt undo data" if its not since we rely on this property below.